build-gradle-plugin/src/main/kotlin/cz/ackee/gradle/AckeePluginKotlin.kt

```diff
@@ -166,6 +173,15 @@ class AckeePluginKotlin : Plugin<Project> {
                         }
                     }
                 }
+
+                /**
+                 * Run "lint$BuildVariant" task before every "assemble$BuildVariant" tasks
+                 */
+                android.applicationVariants.forEach { variant ->
```

Comment:
Should be using `all` from `org.gradle.api.DomainObjectCollection` instead of `forEach` from `kotlin.collections`.

```kotlin
    /**
     * Executes the given action against all objects in this collection, and any objects subsequently added to this
     * collection.
     *
     * @param action The action to be executed
     */
    void all(Action<? super T> action);
```

The same issue is on lines 90, 94 and 138.

Reply:
Mind that with `all` elements are accessed as `this` instead of `it`.

To bridge from `forEach` to `all`, I typically do something like

```kotlin
                android.applicationVariants.all {
                    val variant = this

                    val apkFile = File(outputs, "App.apk")

                    variant.outputs.all { 
                        val output = this
                        ...
```

Reply:
But why is `all` better? Because if you had used `all` instead of `forEach` then it does not matter when this code is executed. With `forEach` you are kind of forced to use `project.afterEvaluate` which is often unnecessary and can create kind of "race conditions" where one `afterEvaluate` block needs to be run after another `afterEvaluate` block and the other way around, creating kind of cyclic dependency.